Completed homework 15 sprint

RURURuina opened the pull request:
Огромная просьба, если здесь есть косяки, пожалуйста объясните по подробнее
До куратора очень сложно дописаться
Заранее спасибо за ревью! <3

RURURuina committed:
Completed homework 15 sprint

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker/data/repositories/MediaPlayerRepository.kt:
```diff
@@ -0,0 +1,10 @@
+package com.practicum.playlistmaker.data.repositories
+
+interface MediaPlayerRepository {
```
Интерфейс репозитория нужно в домейне хранить, чтобы инвертировать зависимость, т.е. чтобы домейн не зависел от data, а data зависел от domain, а сам домейн оставался чистым

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker/presentation/activities/AudioPlayerActivity.kt:
```diff
@@ -12,30 +12,31 @@ import androidx.appcompat.app.AppCompatActivity
-import com.practicum.playlistmaker.SearchActivity.Companion.AUDIO_PLAYER_KEY
+import com.practicum.playlistmaker.R
+import com.practicum.playlistmaker.data.repositories.MediaPlayerRepositoryImpl
+import com.practicum.playlistmaker.data.repositories.MediaPlayerRepository
```
Сейчас есть зависимость этого слоя от data, что нарушает правило зависимостей. Структура должна быть следующей:
1. В домейн слое должен быть интерфейс интерактора и его реализация (которая использует интерфейс репозитория), интеракторы являются основой бизнес логики приложения
2. Интерфейс репозитория тоже должен быть в домейн слое, а его реализация - в data
3. Рекомендуется ещё создать объект Creator где-нибудь в корне проекта или в отдельном пакете, чтобы слои не знали о реализациях, а использовали только интерфейсы (о конкретных реализациях будет знать только Creator, пример такого объекта есть в теории курса)

Таким образом, активити будет работать только с интерфейсом интерактора, тот в свою очередь будет проксировать вызовы в репозиторий

По зависимостям должна получиться следующая схема:
1. presentation зависит от domain слоя
2. data зависит от domain слоя
3. domain слой не зависит ни от кого

ArturNurtdinov reviewed (commented):
В целом работа очень хорошая, молодец, немного только нужно поправить зависимости между слоями, а почти все основные компоненты ты уже создал (не забудь интерактор)

RURURuina committed:
Completed homework 15 sprint
1. Добавлен интерфейс интерактора и его реализация
2. Починил зависимости
3. Добавлен Creator в корне проекта
